packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts

```diff
@@ -231,6 +248,76 @@ export class PluginLedgerConnectorBesu
     }
   }
 
+  public async invokeContractV2(
+    req: InvokeContractV2Request,
+  ): Promise<InvokeContractV2Response> {
+    const fnTag = `${this.className}#invokeContractV2()`;
+    const contractName = req.contractName;
+
+    if (Object.keys(this.contracts).length === 0) {
+      if (this.contractsPath != undefined) {
+        const networkId = await this.web3.eth.net.getId();
+        const contractJson = JSON.parse(
+          fs.readFileSync(this.contractsPath, { encoding: "utf-8" }),
```

Comment:
@jordigiam Please open an GH issue to eliminate the mandatory file system dependency here (it's cool as an optional way of loading the contracts but it should also support pulling them from either the consortium database or the keychain plugin or just from a plain parameter on the constructor options object).

Reply:
Sure!